src/core/plugins/configs/actions.js

```diff
@@ -19,8 +24,35 @@ export function toggle(configName) {
   }
 }
 
-
 // Hook
 export const loaded = () => () => {
   // noop
 }
+
+export const downloadConfig = (req) => (system) => {
+  const {
+    fn: { fetch },
+  } = system
+
+  return fetch(req)
+}
+
+export const getConfigByUrl =
```

Comment:
If we call `parseConfig` with system as seconds argument the possible error thrown inside it will get processed by system.

```js
export const getConfigByUrl =
  (req, cb) =>
  (system) => {
    const { specActions, configsActions } = system

    if (req) {
      return configsActions.downloadConfig(req).then(next, next)
    }

    function next(res) {
      if (res instanceof Error || res.status >= 400) {
        specActions.updateLoadingStatus("failedConfig")
        specActions.updateLoadingStatus("failedConfig")
        specActions.updateUrl("")
        console.error(res.statusText + " " + req.url)
        cb(null)
      } else {
        cb(parseYamlConfig(res.text, system))
      }
    }
  }
```